src/test/java/integration/ScreenFactoryTest.java

```diff
@@ -0,0 +1,58 @@
+package integration;
+
+import aquality.appium.mobile.application.AqualityServices;
+import aquality.appium.mobile.application.MobileModule;
+import aquality.appium.mobile.screens.AndroidScreen;
+import aquality.appium.mobile.screens.IOSScreen;
+import integration.screens.IFakeLoginScreen;
+import integration.screens.ILoginScreen;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+public class ScreenFactoryTest {
+
+    private static final String PLATFORM_NAME_VARIABLE_NAME = "platformName";
+    private static final String PACKAGE_NAME_WITH_SCREENS_VARIABLE_NAME = "packageNameWithScreens";
+
+    @Test
+    public void shouldBePossibleToGetScreenViaFactory() {
+        Assert.assertNotNull(AqualityServices.getScreenFactory().getScreen(ILoginScreen.class));
+    }
+
+    @Test
+    public void shouldBePossibleToGetAndroidScreenViaFactory() {
+        System.setProperty(PLATFORM_NAME_VARIABLE_NAME, "android");
+        AqualityServices.initInjector(new MobileModule(AqualityServices::getApplication));
+        Assert.assertTrue(AqualityServices.getScreenFactory().getScreen(ILoginScreen.class) instanceof AndroidScreen);
+    }
+
+    @Test
+    public void shouldBePossibleToGetIOSScreenViaFactory() {
+        System.setProperty(PLATFORM_NAME_VARIABLE_NAME, "ios");
+        AqualityServices.initInjector(new MobileModule(AqualityServices::getApplication));
+        Assert.assertTrue(AqualityServices.getScreenFactory().getScreen(ILoginScreen.class) instanceof IOSScreen);
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionOnNotImplementedScreenInterface() {
+        Assert.assertThrows(IllegalArgumentException.class, () -> AqualityServices.getScreenFactory().getScreen(IFakeLoginScreen.class));
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionOnNotValidPackageNameWithScreensValue()
+    {
+        System.setProperty(PACKAGE_NAME_WITH_SCREENS_VARIABLE_NAME, "fake.package");
+        AqualityServices.initInjector(new MobileModule(AqualityServices::getApplication));
+        Assert.assertThrows(IllegalArgumentException.class, () -> AqualityServices.getScreenFactory().getScreen(ILoginScreen.class));
+    }
+
+    @AfterClass
+    public void afterAll() {
+        System.clearProperty(PLATFORM_NAME_VARIABLE_NAME);
```

Comment:
I've reviewed test run (https://dev.azure.com/aquality-automation/aquality-automation/_build/results?buildId=1047&view=results) and from error messages and logs can see that the wrong profile for web tests was used. So you've probably broken this with setting/resetting of respectice environment variables.
Please review this.

Reply:
You know, it is quite weired that env variable, that cleared after tests can affect other tests. So, the problem probably not in these tests.